test/framework-ondata-mocha.js

```diff
@@ -1048,102 +1063,159 @@ should.be.rejectedWith('No wrapper fRunUpdater function found!  Cannot run updat
+                                                     },
+                                                     {fRunUpdater: function(vni, args) { 
+
+                                                         nodeInstance = vni.nodeInstance;
+                                                         nodeInstance.vnis.should.be.an('object');
+                                                         nodeInstance.vnis.should.have.all.keys('', '1'); 
+
+                                                         var inputStates = vni.inputStates();
+                                                         inputStates.should.have.all.keys('input', 'iip');
+
+                                                         test.verifyState(inputStates.input, '1', initData);
+                                                         test.verifyState(inputStates.iip, '', iipData);
+
+                                                         vni.outputState({vnid:'1',
+                                                                          data: inputStates.input.data + 
+                                                                                ' ' + inputStates.iip.data + 
+                                                                                ' ' + targetData,
+                                                                          lm: createLm()});    
                                                  }}
                 )}
             }).then(function(network){
-
-                 var testNode = network.processes.testNode.component;
+                 var targetNode = network.processes.targetNode.component;
+                 network.graph.addEdge('initNode', 'output', 'targetNode', 'input');
 
                  return new Promise(function(done, fail) {
+                     test.onOutPortData(targetNode, 'output', done);
 
-		     test.onOutPortData(testNode, 'output', done);
+                     network.graph.addInitial(iipData, 'targetNode', 'iip');
+                     network.graph.addInitial(initData, 'initNode', 'input');
 
-                     network.graph.addInitial(5, 'testNode', 'input');
                  }).then(function(done) {
-                     test.verifyState(done, '', testData);
+                     // Verify we got the correct state which drew from both IIP and non IIP state data
+                     test.verifyState(done, '1', 'Salve! Come va? Addio!');
 
-                     // Verify default VNI has been cleared
-                     expect(nodeInstance).to.not.be.empty;
-                     nodeInstance.vnis.should.be.an('object');
-                     var vnids = Object.keys(nodeInstance.vnis);
-                     vnids.should.not.contain(''); // no default VNI
-                     vnids.should.have.length(5); // we should still have 5 other VNIs that were created
+                     // Now verify that the non IIP input state was cleared by the framework
```

Comment:
I am confused here.  You might have to walk me through this.  If  am understanding correctly, we are on the targetNode at this point, and updater has fired under vnid 1 (but not vnid '').  If so, then shouldn't VNI 1 be deleted? And VNI '' should be unaffected, right?